plugins/user_mapping/src/interface.h

```diff
@@ -0,0 +1,53 @@
+#ifndef IRODS_HTTP_API_USER_MAPPER_INTERFACE_H
+#define IRODS_HTTP_API_USER_MAPPER_INTERFACE_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/// Initializes the user mapping plugin.
+///
+/// \param[in] _args A C-string containing the JSON representing the configuration for the plugin.
```

Comment:
The configuration is 'free-form', in that any plugin can ask for any valid JSON for the configuration. Don't think there's a way to document that structure...

Even the two plugins we have right now ask for different things.

Reply:
Oh, I see. We don't really know what we're going to get... Maybe just mention that the form of the JSON depends on what is provided by the OpenID provider and that should be enough. Thoughts?

Reply:
Since this is only based on the specific plugin used, we could append `The structure of the JSON depends on the requirements specified by the plugin`?

Would this also make it clear to plugin developers that they need to specify the structure required, or would this throw them off?

Reply:
> Would this also make it clear to plugin developers that they need to specify the structure required, or would this throw them off?

I think it would be helpful to mention to plugin developers that they should document the required JSON structure, yes. It may not be particularly useful (since it should be mostly "automatic") but it will help future readers understand what is required by the plugin.

Reply:
an interesting thought.... should we provide a way for a plugin developer to also 'ship a schema'?   and have the plugin be able to validate it's own configuration?   not sure exactly where the dotted line is between docs and schema and responsibility of the developer... but could be an interesting though experiment for a plugin interface.

Reply:
That is already supported IMO. The plugin interface supports an init function and  developers are free to do whatever they want inside the implementation as long as it follows the requirements of the plugin interface.

Developers writing custom plugins should follow normal software development practices, which includes documentation. I don't think we have to state that.

Reply:
good good.

Reply:
Added the line I gave as an example.

Reply:
To add to the 'provide a validator for other plugins' discussion, if we do use the separate template repo, or provide a template in general, maybe including it there might be of interest.

Gonna resolve this though.